src/lib/tauri/http/Response.ts

```diff
@@ -0,0 +1,51 @@
+import * as http from '@tauri-apps/api/http';
+import { either, flow, iots } from '@code-expert/prelude';
+import { HttpError, httpError } from './HttpError';
+
+const ResponseError = iots.strict({
+  statusCode: iots.number,
+  error: iots.string,
+  message: iots.string,
+});
+type ResponseError = iots.TypeOf<typeof ResponseError>;
+
+const Response = iots.union([
+  iots.strict({
+    ok: iots.literal(true),
+    data: iots.unknown,
+  }),
+  iots.strict({
+    ok: iots.literal(false),
+    data: ResponseError,
+  }),
+]);
+
+export const parseResponse: <A>(
+  codec: iots.Decoder<unknown, A>,
+) => (r: http.Response<unknown>) => either.Either<HttpError, A> = (codec) =>
+  flow(
+    Response.decode,
+    either.mapLeft(iots.formatValidationErrors),
+    either.getOrThrow(
+      (errs) =>
+        new Error(`Fetch response did not match Tauri’s "Response" type': ${errs.join('; ')}`),
```

Comment:
"Tauri's Response type" has fastify's Error Response type baked in. In case we eventually need to parse responses from other resources where the error responses don't stem from tauri we might not want to throw here already. Or move the error parsing into httpSigned

Reply:
but it's probably easier to clean up these errors in your Exception-Handling branch